docs/a-walk-through-vhs.md

```diff
@@ -0,0 +1,250 @@
+
+```html
+<video-js id="myPlayer" class="video-js" data-setup='{}'>
+  <source src="http://example.com/manifest.m3u8" type="application/x-mpegURL">
+</video-js>
+```
+
+Looking at the `<source>` tag, `VhsSourceHandler` already used the "type" to tell Video.js that it could handle the source. `VhsHandler` took the manifest URL, in this case "manifest.m3u8" and provided it to the constructor of MPC.
+
+The first thing that MPC must do is download that source, but it doesn't make the request itself. Instead, it creates [this.masterPlaylistLoader_](https://github.com/videojs/http-streaming/blob/0964cb4827d9e80aa36f2fa29e35dad92ca84111/src/master-playlist-controller.js#L264-L266).
+
+```mermaid
+flowchart TD
+  VhsSourceHandler --> VhsHandler
+  VhsHandler --> MasterPlaylistController
+  MasterPlaylistController --> PlaylistLoader
+```
+
+`masterPlaylistLoader_` is an instance of either the [HLS PlaylistLoader](https://github.com/videojs/http-streaming/blob/0964cb4827d9e80aa36f2fa29e35dad92ca84111/src/playlist-loader.js#L379) or the [DashPlaylistLoader](https://github.com/videojs/http-streaming/blob/0964cb4827d9e80aa36f2fa29e35dad92ca84111/src/dash-playlist-loader.js#L259).
+
+The names betray their use. They load the playlist. The URL ("manifest.m3u8" here) is given, and the manifest/playlist is downloaded and parsed. If the content is live, the playlist loader also handles refreshing the manifest. For HLS, where manifests point to other manifests, the playlist loader requests those as well.
+
+As for parsing, for HLS, the manifest responses are parsed using [m3u8-parser](https://github.com/videojs/m3u8-parser). For DASH, the manifest response is parsed using [mpd-parser](https://github.com/videojs/mpd-parser). The output of these parsers is a JSON object that VHS understands. The main structure can be seen in the READMEs, e.g., [here](https://github.com/videojs/m3u8-parser#parsed-output).
+
+So what was once a URL in a `<source>` tag was requested and parsed into a JSON object like the following:
+
+```
+Manifest {
+  playlists: [
+    {
+      attributes: {},
+      Manifest
+    }
+  ],
+  mediaGroups: { ... },
+  segments: [ ... ],
+  ...
+}
+```
+
+Many properties are removed for simplicity. This is a top level manifest (often referred to as a master or main manifest), and within it there are playlists, each playlist being a Manifest itself. Since the JSON "schema" for main and media playlists is the same, you will see irrelevant properties within any given manifest object. For instance, you might see a `targetDuration` property on the main manifest object, though a main manifest doesn't have a target duration. You can ignore irrelevant properties. Eventually they should be cleaned up, and a proper schema defined for manifest objects.
```

Comment:
There's some room for confusion here. We have names for the master manifest (master, main) and then names for the playlists, which are subsequently referred to as "media playlists". Might be room to tighten up the phrasing here.